File: backend/Models/Data/HotSpot.cs
namespace WhaleSpotting.Models.Data;

public class HotSpot
{
    public int Id { get; set; }

    public required string Name { get; set; }

    public required decimal Latitude { get; set; }

    public required decimal Longitude { get; set; }

    public required string Country { get; set; }

    public List<ViewingSuggestion> ViewingSuggestions { get; set; } = null!;
}

File: frontend/src/models/view/HotSpot.ts
import Species from "./Species"

interface ViewingSuggestion {
  id: number
  hotSpotId: number
  speciesId: number
  species: Species
  platforms: string
  platformBoxes: number[]
  timeOfYear: string
  months: number[]
}

interface HotSpot {
  id: number
  name: string
  latitude: number
  longitude: number
  country: string
  viewingSuggestions: ViewingSuggestion[]
}

export default HotSpot
